src/main/java/duke/Tasks/Task.java

```diff
@@ -0,0 +1,177 @@
+            String task = description.replace("event ", "");
+            String[] splitStart = task.split(" /from ");
+            if (splitStart.length == 1) {
+                throw new EmptyDetailsOfTaskError("The start of a event cannot be empty.");
+            }
+            String[] splitEnd = splitStart[1].split(" /to ");
+            if (splitEnd.length == 1) {
+                throw new EmptyDetailsOfTaskError("The end of a event cannot be empty.");
+            }
+            String taskName = splitStart[0];
+            String start = splitEnd[0];
+            String end = splitEnd[1];
+            Task currentTask = new Events(taskName, start, end);
+            return currentTask;
+        } else if (splittedDescription.length == 1 &&
+                (commandName.equals(Commands.todo.name()) ||
+                        commandName.equals(Commands.deadline.name()) ||
+                        commandName.equals(Commands.event.name()))) {
+            throw new EmptyDetailsOfTaskError("The description of a " + commandName + " cannot be empty.");
+        } else {
+            throw new UnknownCommandException("I'm sorry, but I don't know what that means :-C");
+        }
+    }
+
+    /**
+     * Takes in the string of the task that is saved in the previously
+     * saved file and returns a task to be added back into the taskList
+     * 
+     * @param data is the string of the task
+     * @return a Task to be added into the taskList
+     */
+    public static Task createTaskFromSavedData(String data) {
+        String taskType = data.substring(2, 3);
+        String taskStatus = data.substring(6, 7);
+        String taskDescription = data.substring(10);
+        Task currentTask = null;
+        if (taskType.equals("T")) {
+            currentTask = new Todo(taskDescription);
+        } else if (taskType.equals("D")) {
+            String[] splittedData = taskDescription.split(Pattern.quote(" (by: "));
+            String taskEnd = splittedData[1].replace(")", "");
+            currentTask = new Deadlines(splittedData[0], taskEnd);
+        } else if (taskType.equals("E")) {
+            String[] splittedData = taskDescription.split(Pattern.quote(" (from: "));
+            String[] splittedDetails = splittedData[1].split(Pattern.quote(" to: "));
+            String taskStart = splittedDetails[0];
+            String taskEnd = splittedDetails[1].replace(")", "");
+            currentTask = new Events(splittedData[0], taskStart, taskEnd);
+        }
+        if (taskStatus.equals("x")) {
+            currentTask.setAsDone();
+        }
+        return currentTask;
+    }
+
+    /**
+     * Converts the date in the string format to a date in
+     * the localDate format
+     * 
+     * @param sDate is the string of the date
+     * @return a localDate object
+     */
+    public static LocalDate convertDatePlease(String sDate) {
```

Comment:
Perhaps this method name should be phrased more as a verb?

Reply:
I agree. I suggest removing the "Please" and phrasing it such as "convertDateFromStringToObj"?